providers/src/airflow/providers/edge/worker_api/routes/logs.py

```diff
@@ -0,0 +1,133 @@
+from airflow.providers.edge.models.edge_logs import EdgeLogsModel
+from airflow.providers.edge.worker_api.auth import jwt_token_authorization_rest
+from airflow.providers.edge.worker_api.datamodels import PushLogsBody, WorkerApiDocs
+from airflow.providers.edge.worker_api.routes._v2_compat import (
+    AirflowRouter,
+    Body,
+    Depends,
+    create_openapi_http_exception_doc,
+    status,
+)
+from airflow.utils.session import NEW_SESSION, create_session, provide_session
+
+logs_router = AirflowRouter(tags=["Logs"], prefix="/logs")
+
+
+@cache
+@provide_session
+def _logfile_path(task: TaskInstanceKey, session=NEW_SESSION) -> str:
+    """Elaborate the (relative) path and filename to expect from task execution."""
+    from airflow.utils.log.file_task_handler import FileTaskHandler
+
+    ti = TaskInstance.get_task_instance(
+        dag_id=task.dag_id,
+        run_id=task.run_id,
+        task_id=task.task_id,
+        map_index=task.map_index,
+        session=session,
+    )
+    if TYPE_CHECKING:
+        assert ti
+    return FileTaskHandler(".")._render_filename(ti, task.try_number)
+
+
+@logs_router.get(
+    "/logfile_path/{dag_id}/{task_id}/{run_id}/{try_number}/{map_index}",
+    dependencies=[Depends(jwt_token_authorization_rest)],
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+        ]
+    ),
+)
+def logfile_path(
+    dag_id: Annotated[str, WorkerApiDocs.dag_id],
+    task_id: Annotated[str, WorkerApiDocs.task_id],
+    run_id: Annotated[str, WorkerApiDocs.run_id],
+    try_number: Annotated[int, WorkerApiDocs.try_number],
+    map_index: Annotated[int, WorkerApiDocs.map_index],
+) -> str:
+    """Elaborate the path and filename to expect from task execution."""
+    task = TaskInstanceKey(
+        dag_id=dag_id, task_id=task_id, run_id=run_id, try_number=try_number, map_index=map_index
+    )
+    return _logfile_path(task)
+
+
+@logs_router.post(
+    "/push/{dag_id}/{task_id}/{run_id}/{try_number}/{map_index}",
+    dependencies=[Depends(jwt_token_authorization_rest)],
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+        ]
+    ),
+)
+def push_logs(
+    dag_id: Annotated[str, WorkerApiDocs.dag_id],
+    task_id: Annotated[str, WorkerApiDocs.task_id],
+    run_id: Annotated[str, WorkerApiDocs.run_id],
+    try_number: Annotated[int, WorkerApiDocs.try_number],
+    map_index: Annotated[int, WorkerApiDocs.map_index],
+    body: Annotated[
+        PushLogsBody,
+        Body(
+            title="Log data chunks",
+            description="The worker remote has no access to log sink and with this can send log chunks to the central site.",
+        ),
+    ],
+) -> None:
+    """Push an incremental log chunk from Edge Worker to central site."""
+    with create_session() as session:
```

Comment:
Why not use `dep_session`?

Reply:
I had that initially and got some weired error from FastAPI that it could not parse the query string. Gave-up after one hour back.and forth.
...but now as Airflow 3 is broken atm anyway, can not re-produce w/o a task running that pushes logs. So cleaned.up. Will need to check once Airflow3 code gets working again.